web/templates/invite_user_modal.hbs

```diff
@@ -46,46 +46,8 @@
             <p class="custom-time-input-formatted-description"></p>
         </div>
     </div>
-    <div class="input-group">
-        <label for="invite_as" class="modal-field-label">{{t "Users join as" }}
-            {{> help_link_widget link="/help/user-roles" }}
-        </label>
-        <select id="invite_as" name="invite_as" class="invite-user-select modal_select bootstrap-focus-style">
-            <option value="{{ invite_as_options.guest.code }}">{{t "Guests" }}</option>
-            <option selected="selected" value="{{ invite_as_options.member.code }}">{{t "Members" }}</option>
-            {{#if is_admin}}
-            <option value="{{ invite_as_options.moderator.code }}">{{t "Moderators" }}</option>
-            <option value="{{ invite_as_options.admin.code }}">{{t "Administrators" }}</option>
-            {{/if}}
-            {{#if is_owner}}
-            <option value="{{ invite_as_options.owner.code }}">{{t "Owners" }}</option>
-            {{/if}}
-        </select>
-    </div>
-    <div class="input-group">
-        <label>{{t "Channels they should join" }}</label>
-        <div id="streams_to_add">
-            {{#if show_select_default_streams_option}}
-            <div class="select_default_streams">
-                <label class="checkbox display-block modal-field-label">
-                    <input type="checkbox" id="invite_select_default_streams" checked="checked" />
-                    <span class="rendered-checkbox"></span>
-                    {{t 'Default channels for this organization'}}
-                </label>
-            </div>
-            {{/if}}
-            <div id="invite_streams_container" class="add_streams_container">
-                <div class="pill-container stream_picker">
-                    <div class="input" contenteditable="true"
-                      data-placeholder="{{t 'Add channels' }}">
-                        {{~! Squash whitespace so that placeholder is displayed when empty. ~}}
-                    </div>
-                </div>
-            </div>
-        </div>
-    </div>
-    <div id="guest_visible_users_container" class="input-group" style="display: none;">
-    </div>
+    {{> user_roles_dropdown invite_as_options=invite_as_options is_admin=is_admin is_owner=is_owner }}
+    {{> streams_to_add show_select_default_streams_option=show_select_default_streams_option}}
```

Comment:
I think it can just be a single file named like `invited_user_configuration_options` which can contain UI for all role, streams and groups and groups can be shown only for invite creation UI.